### PR TITLE
HMAN-57 establish DELETE linkedHearingGroup end-point and validate

### PR DESCRIPTION
**Pull request by rafalkalita**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-57

### Change description ###
Establish the DELETE /linkedHearingGroup/{id} end-point for deleting the members of a group of linked hearing requests. This implementation does the validation, but not invoke the DB delete and does not send any request to List Assist via HMI

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedGroupDetailsRepository.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.hmc.data.LinkedGroupDetails;
+
+@Transactional(propagation = Propagation.REQUIRED)
+@Repository
+public interface LinkedGroupDetailsRepository extends CrudRepository<LinkedGroupDetails, Long> {
+
+    // TODO: implement DB query - https://tools.hmcts.net/jira/browse/HMAN-96
+    default void deleteHearingGroup(Long hearingGroupId) {
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedGroupDetailsRepository.java:14]: is this a place holder for the next coming work?

File: src/integrationTest/java/uk/gov/hmcts/reform/hmc/repository/LinkedGroupDetailsRepositoryIT.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private static final String INSERT_LINKED_HEARINGS_DATA_SCRIPT = "classpath:sql/insert-linked-hearings.sql";
+    private static final String DELETE_HEARING_DATA_SCRIPT = "classpath:sql/delete-hearing-tables.sql";
+
+    @Autowired
+    LinkedGroupDetailsRepository linkedGroupDetailsRepository;
+
+    @Test
+    @Sql(scripts = {DELETE_HEARING_DATA_SCRIPT, INSERT_LINKED_HEARINGS_DATA_SCRIPT})
+    void testFindByHearingGroupId() {
+        Optional<LinkedGroupDetails> linkedGroupDetailsOptional = linkedGroupDetailsRepository.findById(7600000000L);
```
Comment (kiran-yenigala-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/hmc/repository/LinkedGroupDetailsRepositoryIT.java:24]: do we really need a test case for this, as findById is not overrider or changed the implementation in our code

File: src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupService.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            .max(Map.Entry.comparingByKey());
+
+        return max.isPresent() ? max.get().getValue() : List.of();
+    }
+
+    private void deleteFromLinkedGroupDetails(List<HearingEntity> linkedGroupHearings, Long hearingGroupId) {
+        linkedGroupHearings.forEach(hearingEntity -> {
+            // TODO: unlink hearingEntity from the group and persist - https://tools.hmcts.net/jira/browse/HMAN-96
+        });
+        linkedGroupDetailsRepository.deleteHearingGroup(hearingGroupId);
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupService.java:121]: as this method is not implemented in repository, how we are going to test this logic

Comment (rafalkalita) [reply src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupService.java:121]: This has been added to the repository, so I could mock it in the service/ verify it has been invoked. And to test that if  the exception occurs it should not be invoked.

File: src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedGroupDetailsRepository.java (hunk 1 above)
Comment (rafalkalita) [reply src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedGroupDetailsRepository.java:14]: yes, this ticket is just to establish the endpoint and do the validation

**Pull request merged**